src/programs.rkt

```diff
@@ -139,6 +139,46 @@
     (for/list ([root (in-list roots)])
       (vector-ref vregs root))))
 
+(define (make-compiler arg->precision operator-handler operator-type-handler
+                       if-handler if-type interpreter-name)
```

Comment:
Names are a tad confusing. Perhaps `operator-proc` for `operator-handler`, `operator-itypes` for `operator-type-handler`, `if-proc` for `if-handler` and `cond-type` for `if-type`.